db/view/view.cc

```diff
@@ -648,12 +731,17 @@ void view_updates::create_entry(const partition_key& base_key, const clustering_
     if (!matches_view_filter(*_base, _view_info, base_key, update, now)) {
         return;
     }
-    deletable_row& r = get_view_row(base_key, update);
-    auto marker = compute_row_marker(update);
-    r.apply(marker);
-    r.apply(update.tomb());
-    add_cells_to_view(*_base, *_view, row(*_base, column_kind::regular_column, update.cells()), r.cells());
-    _op_count++;
+
+    auto view_rows = get_view_rows(base_key, update, std::nullopt);
+    auto update_marker = compute_row_marker(update);
+    for (const auto& [r, action]: view_rows) {
+        if (!action.apply_row_marker(r, now)) {
```

Comment:
self note: I suffered from the same condition, I'll start reviewing this one tomorrow in the morning, it's too complicated for my afternoon brain

Reply:
Ok, I tried following what happens here, and am I correct that `apply_row_marker` is a helper function which does the following:
 - if the `action` variant stores `row_marker`, it will be applied and return true
 - otherwise it returns false
?

If that's the case, this code is very not obvious to the reader, it took me a longer while to decipher what happens here. `action` is already a variant, so isn't this code equivalent to:
```cpp
if (auto marker = std::get_if<row_marker>(&action._action) {
    r->apply(*marker, now);
}
```
? I think it isn't more bloated than current code, but I find it much more self-descriptive. When the `visit` is hidden under a helper function which returns true when the action matched, and false otherwise, it's much less clear.